internal/server.go

```diff
@@ -481,6 +481,77 @@ func (s *Server) DiscoverSchemas(ctx context.Context, req *pub.DiscoverSchemasRe
 	}, err
 }
 
+func (s *Server) DiscoverRelatedEntities(ctx context.Context, req *pub.DiscoverRelatedEntitiesRequest) (*pub.DiscoverRelatedEntitiesResponse, error) {
```

Comment:
Needs to be modified to loop through the input schemas on the request and only return related entities to the ones requested.